### PR TITLE
Modernize the declaration as a hybrid UMD + explicit global

### PR DESCRIPTION
**Pull request by aluanhaddad**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by aluanhaddad** [comment]
I feel like the version number should be updated since this is potentially breaking but that would cause the version number to become out of sync with systemjs itself.

File: types/systemjs/index.d.ts [hunk 1]
```diff
@@ -1,11 +1,26 @@
 // Type definitions for SystemJS 0.20
 // Project: https://github.com/systemjs/systemjs
-// Definitions by: Ludovic HENIN <https://github.com/ludohenin>, Nathan Walker <https://github.com/NathanWalker>, Giedrius Grabauskas <https://github.com/GiedriusGrabauskas>
+// Definitions by: Ludovic HENIN <https://github.com/ludohenin>, Nathan Walker <https://github.com/NathanWalker>, Giedrius Grabauskas <https://github.com/GiedriusGrabauskas>,
+//                 Aluan Haddad <https://github.com/aluanhaddad>
```
Comment (GiedriusGrabauskas) [inline types/systemjs/index.d.ts]: Split to new lines.
https://github.com/DefinitelyTyped/DefinitelyTyped#edit-an-existing-package

File: types/systemjs/index.d.ts [hunk 2]
```diff
@@ -248,25 +263,16 @@ declare namespace SystemJSLoader {
         };
     }
 
-    interface SystemJSSystemFields {
```
Comment (GiedriusGrabauskas) [inline types/systemjs/index.d.ts]: Why did you remove this interface?

File: types/systemjs/index.d.ts [hunk 3]
```diff
@@ -288,7 +294,6 @@ declare namespace SystemJSLoader {
          * Returns a module from the registry by normalized name.
          */
         get(moduleName: string): any;
```
Comment (GiedriusGrabauskas) [inline types/systemjs/index.d.ts]: Missing generic value, use defaults:
```ts
get<TModule = any>(moduleName: string): TModule;
```

File: types/systemjs/index.d.ts [hunk 4]
```diff
@@ -305,7 +310,6 @@ declare namespace SystemJSLoader {
          * Promise resolves to the module value.
          */
         import(moduleName: string, normalizedParentName?: string): Promise<any>;
```
Comment (GiedriusGrabauskas) [inline types/systemjs/index.d.ts]: Same like a `get` method, please add a missing generic value.

File: types/systemjs/index.d.ts [hunk 5]
```diff
@@ -318,7 +322,6 @@ declare namespace SystemJSLoader {
          * Useful when writing a custom instantiate hook or using System.set.
          */
         newModule(object: any): any;
```
Comment (GiedriusGrabauskas) [inline types/systemjs/index.d.ts]: Same, add missing generic value.

**Review by GiedriusGrabauskas — CHANGES_REQUESTED** [review]
About the version, yes, you should leave it, because `SystemJS` version is still `0.20` (newest `0.20.19`).

File: types/systemjs/index.d.ts (hunk 5 above)
Comment (aluanhaddad) [reply types/systemjs/index.d.ts]: All uses of generics were raising a dtslint error. Since the generics are not associated with either the parameters or the return values of the declaring functions, they essentially behave like a type assertion.

The rule is explained here. https://github.com/Microsoft/dtslint/blob/master/docs/no-unnecessary-generics.md

File: types/systemjs/index.d.ts (hunk 2 above)
Comment (aluanhaddad) [reply types/systemjs/index.d.ts]: I removed it because it feels awkward. The _fields_ suffix is unusual, properties would be more standard but even then the segregation is not very unidiomatic. Overall I felt it would be more straightforward if the fields were inlined.

File: types/systemjs/index.d.ts (hunk 4 above)
Comment (aluanhaddad) [reply types/systemjs/index.d.ts]: All uses of generics were raising a dtslint error. Since the generics are not associated with either the parameters or the return values of the declaring functions, they essentially behave like a type assertion.

The rule is explained here. https://github.com/Microsoft/dtslint/blob/master/docs/no-unnecessary-generics.md

**Review by GiedriusGrabauskas — APPROVED** [review]
Okay, then merge will be with breaking changes. 👍
